### DISCUSSION TITLE
[@types/ember-feature-flags] Does not export types for helpers/feature-flag

### DISCUSSION THREAD

**Discussion by dsafonov-grid** [DefinitelyTyped/DefinitelyTyped#60167] [Issues with a @types package]
For our application purposes we have to re-export this helper like this
```js
export { default } from 'ember-feature-flags/helpers/feature-flag';
```
Due to lack of typings, TS highlights the following error
_Cannot find module 'ember-feature-flags/helpers/feature-flag' or its corresponding type declarations._

To calm down TS I want to add typing for this helper like this
<img width="1188" alt="Screen Shot 2022-05-03 at 17 23 22" src="https://user-images.githubusercontent.com/82807060/166460946-5ebada8b-6cb2-4967-bba8-18aa95704a96.png">

[a comment by an automated account is not shown]

**Comment by chriskrycho** [comment]
@dsafonov-grid thanks for suggesting this! I think the overall idea is great, but you might actually consider writing it using a signature *and* avoiding the internal implementation details:

```ts
import Helper from '@ember/component/helper';

interface FeatureFlagsSignature {
  Args: {
    Positional: [string]
  };
};

export default class FeatureFlag extends Helper<FeatureFlagsSignature> {}
```

That will provide the bits you need for direct usage in your own app *and* make it usable for [Glint](https://github.com/typed-ember/glint) users.

**Reply by dsafonov-grid** [reply]
Thank you a lot @chriskrycho for advice. 
I opened PR: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60194

PS not sure how to test this change tbh

**Reply by dsafonov-grid** [reply]
@chriskrycho, could you please clarify, if we have PR like this https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60194 regarding file placement. 

Will I be able to re-export like this? 
```js
export { default } from 'ember-feature-flags/helpers/feature-flag';
```

Unfortunately I struggle with understanding how my typing will be available for importing by others

**Reply by chriskrycho** [reply]
@dsafonov-grid, sorry, this dropped off my radar. You can re-export like that, yes. There’s nothing special about the TS part of this, so reading [the `export` docs](http://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export) and [the `import` docs](http://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) on MDN is a good way to get your head around these things! Ember’s module system has not used the Node resolution algorithm for where to find modules historically, but it *has* always been compliant with the spec for modules, so you can safely rely on the docs to understand it correctly.
